src/domain/model/test/interpretation/InterpretationRules.ts

```diff
@@ -0,0 +1,40 @@
+import { Results } from '../Results';
+import { Condition } from './Condition';
+import { OutputPattern } from './Output';
+import { Interpretation } from './Interpretation';
+import { DomainValidationError } from '../../DomainValidationError';
+
+export class InterpretationRules {
+  constructor(private rules: InterpretationRule[]) {}
+
+  interpret(results: Results): Array<Interpretation> {
+    return this.rules
+      .map((rule) => rule.interpret(results))
+      .filter((interpretation) => interpretation !== null) as Interpretation[];
```

Comment:
Here you are expecting null, slightly confused by that...

Reply:
InterpretationRule.interpret returns `null` if the rule doesn't match.

Reply:
I can split to `matches()` and `getInterpretation()`. Probably would read better

Reply:
It's fairly clear, I was just confused as I had been looking at InterpretationRules, which returns an empty array.